### PR TITLE
#151991933 Implement jwt signup

### PR DESCRIPTION
**Pull request by ChidinmaOrajiaku**
What does this PR do?
Implements jwt on user signup

Description of Task to be completed?
Users once authenticated via the signup api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication.

How should this be manually tested?
Sign up and check local storage for jwt

Any background context you want to provide?
None

What are the relevant pivotal tracker stories?
#151991933 Users should receive a JWT on successful signup

Screenshots (if appropriate)
image

Questions:
Not applicable

### CODE REVIEW

**Review by temilaj — CHANGES_REQUESTED** [review]
Please refactor to using ES6 or greater syntax

File: app/controllers/users.js [hunk 1]
```diff
@@ -72,40 +72,50 @@ exports.checkAvatar = function(req, res) {
...
-            return res.render('/#!/signup?error=unknown', {
-              errors: err.errors,
+exports.create = (req, res, next) => {
+    if (req.body.name && req.body.password && req.body.email) {
+      User.findOne({
+        email: req.body.email
+      }).exec((err,existingUser) => { 
+        if (!existingUser) {
+          const user =  new User({
+            name: req.body.name,
```
Comment (temilaj) [inline app/controllers/users.js]: consider using object destructuring here, and then the property shorthand

File: app/views/users/signup.jade [hunk 2]
```diff
@@ -33,7 +33,7 @@ block auth
 
   div.sign-body
     div.more-sign
-      form.signup.form-horizontal(action="/users", method="post")
+      form.signup.form-horizontal(action="/api/auth/signup", method="post")
```
Comment (temilaj) [inline app/views/users/signup.jade:36]: It's a better practice to version your API tasks

File: public/js/controllers/index.js [hunk 3]
```diff
@@ -15,6 +19,32 @@ angular.module('mean.system')
       }
     };
 
+    const validation = (token) => {
```
[the remark this answers is not in the document]
Comment (temilaj) [reply public/js/controllers/index.js]: Can you think of a better function name than `validation` here?

File: test/user/signup.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  describe('Users', () => {
+    it('should get token on successful sign up', (done) => {
+      testUser = Object.assign({}, user);
+      chai.request(app)
+        .post('/api/auth/signup')
+        .send(testUser)
+        .end((err, res) => {
+          res.should.have.status(201);
+          res.body.should.have.token;
+          res.body.should.have.message;
```
Comment (temilaj) [inline test/user/signup.js:38]: How do you think this test can be better refactored to a more comprehensive test?

**Review by temilaj — CHANGES_REQUESTED** [review]

File: test/user/signup.js (hunk 4 above)
Comment (ChidinmaOrajiaku) [reply test/user/signup.js:38]: Feedback implemented

File: public/js/controllers/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChidinmaOrajiaku) [reply public/js/controllers/index.js]: Feedback implemented

**Pull request closed without merging**